data/data-pipeline/data_pipeline/etl/score/etl_score.py

```diff
@@ -310,61 +316,21 @@ def _add_percentiles_to_df(
 
         else:
             # For agricultural loss, we are using whether there is value at all to determine percentile and then
-            # filling places where the value is False with 0
+            # filling places where the value is False with np.NaN (since it doesn't apply)
+            # For linguistic isolation, we drop PR and then recalculate. We fill PR with np.NaN
             df[
                 f"{output_column_name_root}"
                 f"{field_names.PERCENTILE_FIELD_SUFFIX}"
-            ] = (
-                df.where(
-                    df[field_names.AGRICULTURAL_VALUE_BOOL_FIELD].astype(float)
-                    == 1.0
-                )[input_column_name]
-                .rank(ascending=ascending, pct=True)
-                .fillna(
-                    df[field_names.AGRICULTURAL_VALUE_BOOL_FIELD].astype(float)
-                )
-            )
-
-        # Create the urban/rural percentiles.
-        urban_rural_percentile_fields_to_combine = []
-        for (urban_or_rural_string, urban_heuristic_bool) in [
-            ("urban", True),
-            ("rural", False),
-        ]:
-            # Create a field with only those values
-            this_category_only_value_field = (
-                f"{input_column_name} (value {urban_or_rural_string} only)"
-            )
-            df[this_category_only_value_field] = np.where(
-                df[field_names.URBAN_HEURISTIC_FIELD] == urban_heuristic_bool,
-                df[input_column_name],
-                None,
-            )
-
-            # Calculate the percentile for only this category
-            this_category_only_percentile_field = (
-                f"{output_column_name_root} "
-                f"(percentile {urban_or_rural_string} only)"
-            )
-            df[this_category_only_percentile_field] = df[
-                this_category_only_value_field
+            ] = df.groupby(df[field_names.GEOID_TRACT_FIELD].isin(drop_tracts))[
+                input_column_name
             ].rank(
-                pct=True,
-                # Set ascending to the parameter value.
-                ascending=ascending,
+                ascending=ascending, pct=True
             )
-
-            # Add the field name to this list. Later, we'll combine this list.
-            urban_rural_percentile_fields_to_combine.append(
-                this_category_only_percentile_field
-            )
-
-        # Combine both urban and rural into one field:
-        df[
-            f"{output_column_name_root}{field_names.PERCENTILE_URBAN_RURAL_FIELD_SUFFIX}"
-        ] = df[urban_rural_percentile_fields_to_combine].mean(
-            axis=1, skipna=True
-        )
+            df.loc[
+                df[field_names.GEOID_TRACT_FIELD].isin(drop_tracts),
```

Comment:
Can you add some comments about this as well? I'm confused about what's going on. 

Why are you running the percentile *before* setting the values to `None`, and then afterwards setting it to `None` for those drop_tracts?

Reply:
yes! so it's a groupby percentile, so the percentiles get calculated separately (e.g., the drop tracts get their own percentiles), and then we get rid of the percentiles for the droptracts. The idea here is that we want to preserve the raw data for the drop tracts, even if we don't keep the percentiles. Does that make sense to you?